platform/admin/core/admin-core-impl/src/main/java/org/codice/ddf/admin/core/impl/AdminAlertImpl.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.core.impl;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.management.NotCompliantMBeanException;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.shiro.SecurityUtils;
+import org.codice.ddf.admin.core.api.jmx.AdminAlertMBean;
+import org.codice.ddf.persistence.PersistenceException;
+import org.codice.ddf.persistence.PersistentItem;
+import org.codice.ddf.persistence.PersistentStore;
+import org.codice.ddf.system.alerts.Alert;
+import org.codice.ddf.system.alerts.SystemNotice;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventAdmin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.security.SubjectUtils;
+
+public class AdminAlertImpl extends BasicMBean implements AdminAlertMBean {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AdminAlertImpl.class);
+
+    private final String activeAlertQuery;
+
+    private PersistentStore persistentStore;
+
+    private EventAdmin eventAdmin;
+
+    public AdminAlertImpl(PersistentStore persistentStore, EventAdmin eventAdmin)
+            throws NotCompliantMBeanException, UnknownHostException {
+        super(AdminAlertMBean.class, AdminAlertMBean.OBJECT_NAME);
+        this.persistentStore = persistentStore;
+        this.eventAdmin = eventAdmin;
+        activeAlertQuery = String.format("status='%s' AND hostName='%s'",
+                Alert.ALERT_ACTIVE_STATUS,
+                InetAddress.getLocalHost()
+                        .getHostName());
+    }
+
+    @Override
+    public List<Map<String, Object>> getAlerts() {
+        List<Map<String, Object>> alerts = new ArrayList<>();
+        try {
+            List<Map<String, Object>> results = persistentStore.get("alerts", activeAlertQuery);
```

Comment:
If any of these results contain data that was obtained from a user on the system, it would be possible to execute a XSS attack against an administrator user. We need to escape or encode these results to avoid executing any scripts that they might contain since we are pushing them up to a handlebars template that uses the {{{}}} braces and doesn't automatically escape HTML for us.